### ISSUE TITLE
[Feature] `bind` command `ensure-selection-start`

### ISSUE THREAD

**Issue by patrick-kidger** [fish-shell/fish-shell#10310]
Right now the `bind` command has [its own sublanguage](https://fishshell.com/docs/3.3/cmds/bind.html#special-case-the-escape-character). In particular this has a way to swap the start and end of the selection, with `swap-selection-start-stop`.

It'd be great if there was a way to specify that it was specifically at the start or the stop. The reason I'd find this useful is because I'm putting together a [Helix](https://github.com/helix-editor/helix)-mode for editing commands, as an alternative to the existing vi-mode. I'm proposing adding an `ensure-selection-start` command that ensures the cursor is specifically on the left end of the selection.

Poking around a bit, it looks like this should be doable by adding an entry to this table:

https://github.com/fish-shell/fish-shell/blob/8667ed5c17a0333ad642cea4b5dee1007c43bd7d/src/input.rs#L115

and a short implementation in this file:

https://github.com/fish-shell/fish-shell/blob/8667ed5c17a0333ad642cea4b5dee1007c43bd7d/src/reader.rs#L1

Does this seem reasonable to you?

**Comment by krobelus** [comment]
yeah that sounds reasonable. Could probably also benefit https://github.com/sshilovsky/fish-helix
